test/unit/commands/edge/drivers/package.test.ts

```diff
@@ -1,179 +1,284 @@
-		createWriteStream: jest.fn(),
-	}
-})
-jest.mock('yaml', () => {
 	return {
-		safeLoad: jest.fn(),
+		...originalLib,
+		outputItem: jest.fn(),
 	}
 })
 
 describe('package', () => {
-	const existsSyncMock = fs.existsSync as unknown as jest.Mock<boolean, [fs.PathLike]>
-	const lstatSyncMock = fs.lstatSync as unknown as jest.Mock<fs.Stats, [fs.PathLike]>
-	const readFileSyncMock = fs.readFileSync as unknown as jest.Mock<string, [fs.PathLike]>
-	const readdirSyncMock = fs.readdirSync as unknown as jest.Mock<string[], [fs.PathLike]>
-	const createReadStreamMock = fs.createReadStream as unknown as jest.Mock<fs.ReadStream, [fs.PathLike]>
+	const zipContents = {} as Uint8Array
+	const jsZipMock = JSZip as unknown as jest.Mock<JSZip, []>
+	const pipeMock = jest.fn()
+	const readableStream = {
+		pipe: pipeMock,
+	} as unknown as NodeJS.ReadableStream
+	const generateNodeStreamMock = jest.fn().mockReturnValue(readableStream)
+	const generateAsyncMock = jest.fn().mockReturnValue(zipContents)
+	const mockJSZip = {
+		generateNodeStream: generateNodeStreamMock,
+		generateAsync: generateAsyncMock,
+	} as unknown as JSZip
+
+	const getProjectDirectorySpy = jest.spyOn(packageUtil, 'getProjectDirectory')
+	const processConfigFileSpy = jest.spyOn(packageUtil, 'processConfigFile')
+	const processFingerprintsFileSpy = jest.spyOn(packageUtil, 'processFingerprintsFile')
+	const buildTestFileMatchersSpy = jest.spyOn(packageUtil, 'buildTestFileMatchers')
+	const processSrcDirSpy = jest.spyOn(packageUtil, 'processSrcDir')
+	const processProfilesSpy = jest.spyOn(packageUtil, 'processProfiles')
+
+	const driver = { driverId: 'driver id', version: 'driver version' } as unknown as EdgeDriver
+	const outputItemMock = outputItem as unknown as
+		jest.Mock<Promise<EdgeDriver>, [SmartThingsCommandInterface, CommonOutputProducer<EdgeDriver>, GetDataFunction<EdgeDriver>]>
+	outputItemMock.mockImplementation((_command, _config, actionFunction): Promise<EdgeDriver> => {
+		actionFunction()
+		return Promise.resolve(driver)
+	})
+	const uploadSpy = jest.spyOn(DriversEndpoint.prototype, 'upload').mockResolvedValue(driver)
+
+	const chooseChannelSpy = jest.spyOn(channels, 'chooseChannel').mockResolvedValue('channel id')
+	const assignDriverSpy = jest.spyOn(ChannelsEndpoint.prototype, 'assignDriver')
+		.mockResolvedValue({} as DriverChannelDetails)
+
+	const chooseHubSpy = jest.spyOn(hubs, 'chooseHub').mockResolvedValue('hub id')
+	const installDriverSpy = jest.spyOn(HubsEndpoint.prototype, 'installDriver').mockResolvedValue()
+
+	const logSpy = jest.spyOn(PackageCommand.prototype, 'log').mockImplementation()
 
 	afterEach(() => {
 		jest.clearAllMocks()
 	})
 
-	const mockExistsAndIsFile = (): void => {
-		existsSyncMock.mockReturnValueOnce(true)
-		const statsIsFileMock = jest.fn().mockReturnValueOnce(true)
-		lstatSyncMock.mockReturnValueOnce({ isFile: statsIsFileMock } as unknown as fs.Stats)
+	const mockProjectDirectoryProcessing = (): void => {
```

Comment:
This might be a good thing to discuss in the parking lot. I've been kind of torn between doing more of these in the setup and keeping them more explicit in the tests. Putting them up at the top is nice because you can reduce duplication but it makes the tests a little harder to read because you have to check multiple places to see if the mock was happening which has caused me some grief in the past.